sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/ShareFileResourceTests.cs

```diff
@@ -0,0 +1,512 @@
+            Assert.That(dataAt5, Is.EqualTo(fileContentStream.AsBytes().ToArray()));
+            mock.Verify(b => b.UploadRangeAsync(It.IsAny<HttpRange>(), It.IsAny<Stream>(), It.IsAny<ShareFileUploadRangeOptions>(), It.IsAny<CancellationToken>()),
+                Times.Once());
+            mock.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public async Task CopyFromStreamAsync_Error()
+        {
+            // Arrange
+            Mock<ShareFileClient> mock = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/file"),
+                new ShareClientOptions());
+
+            mock.Setup(b => b.UploadAsync(It.IsAny<Stream>(), It.IsAny<ShareFileUploadOptions>(), It.IsAny<CancellationToken>()))
+                .Throws(new RequestFailedException(status: 404, message: "The specified resource does not exist.", errorCode: "ResourceNotFound", default));
+
+            ShareFileStorageResource storageResource = new ShareFileStorageResource(mock.Object);
+
+            // Act
+            int length = 1024;
+            var data = GetRandomBuffer(length);
+            using (var stream = new MemoryStream(data))
+            {
+                await TestHelper.AssertExpectedExceptionAsync<RequestFailedException>(
+                storageResource.CopyFromStreamInternalAsync(stream, length, false, length),
+                e =>
+                {
+                    Assert.AreEqual("ResourceNotFound", e.ErrorCode);
+                });
+            }
+            mock.Verify(b => b.UploadAsync(It.IsAny<Stream>(), It.IsAny<ShareFileUploadOptions>(), It.IsAny<CancellationToken>()),
+                Times.Once());
+            mock.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public async Task CopyFromUriAsync()
+        {
+            // Arrange
+            Mock<StorageResourceItem> sourceResource = new();
+            sourceResource.Setup(b => b.Uri)
+                .Returns(new Uri("https://storageaccount.file.core.windows.net/container/sourcefile"));
+
+            Mock<ShareFileClient> mockDestination = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/destinationfile"),
+                new ShareClientOptions());
+
+            Uri passedSourceUri = default;
+            HttpRange passedRange = default;
+            HttpRange passedSourceRange = default;
+            mockDestination.Setup(b => b.UploadRangeFromUriAsync(It.IsAny<Uri>(), It.IsAny<HttpRange>(), It.IsAny<HttpRange>(), It.IsAny<ShareFileUploadRangeFromUriOptions>(), It.IsAny<CancellationToken>()))
+                .Callback<Uri, HttpRange, HttpRange, ShareFileUploadRangeFromUriOptions, CancellationToken>((sourceUri, range, sourceRange, options, token) =>
+                {
+                    passedSourceUri = sourceUri;
+                    passedRange = range;
+                    passedSourceRange = sourceRange;
+                })
```

Comment:
Future reference: instead of doing this callback where you assign them to some variables, you can use your mock verify with the expected values (or predicates) instead of using `It.IsAny<>()`. e.g.
```csharp
mockDestination.Verify(b => b.UploadRangeFromUriAsync(
    sourceResource.Object.Uri,
    new HttpRange(0, length),
    new HttpRange(0, length),
    It.IsAny<ShareFileUploadRangeFromUriOptions>(),
    It.IsAny<CancellationToken>()),
    Times.Once());
```
Maybe not really a reduction of code amount but it gets rid of a ton of variables just floating around, making it clear you're explicitly testing if the mock got called with this explicit set of parameters.